### DISCUSSION TITLE
Cannot activate tree-sitter, because language grammar for tsx is unavailable (not-found):

### DISCUSSION THREAD

**Discussion by 99702** [joaotavora/eglot#1232] [Q&A] [answered]
I'm using Emacs 30 , default eglot. When i open a `.ts` file it comes with error saying

```
⛔ Warning (treesit): Cannot activate tree-sitter, because language grammar for typescript is unavailable (symbol-error): /home/kali/.emacs.d/tree-sitter/libtree-sitter-typescript.so: undefined symbol: tree_sitter_typescript
⛔ Warning (treesit): Cannot activate tree-sitter, because language grammar for tsx is unavailable (not-found): (libtree-sitter-tsx libtree-sitter-tsx.0 libtree-sitter-tsx.0.0 libtree-sitter-tsx.so libtree-sitter-tsx.so.0 libtree-sitter-tsx.so.0.0) No such file or directory
⛔ Warning (treesit): Cannot activate tree-sitter, because language grammar for typescript is unavailable (symbol-error): /home/kali/.emacs.d/tree-sitter/libtree-sitter-typescript.so: undefined symbol: tree_sitter_typescript
```
I've already installed  treesitter packages before compilation. 
`sudo dnf install tree-sitter tree-sitter-devel
`

But for javascript or any other language it works.

**Comment by joaotavora** [comment]
Sorry, but this has nothing to do with Eglot. You have to install the language grammars in addition to tree-sitter.  Ask in help-gnu-emacs@gnu.org

**Comment by 99702** [comment, marked as answer]
it was solved. Later by using treesit's, `treesit-install-language-grammer` library for manually adding typescript-ts-mode , with specifiying its parser.c manually solved the problem
